IRescue/Unity/Assets/Scripts/Unity/ObjectPlacing/States/ObjectPlacementState.cs

```diff
@@ -12,12 +15,116 @@ namespace Assets.Scripts.Unity.ObjectPlacing.States
     public class ObjectPlacementState : AbstractState
     {
         /// <summary>
+        ///  The time that will be kept to not immediately place buildings.
+        /// </summary>
+        private long hoverTime;
+
+        /// <summary>
+        /// True if a building is being moved, else false
+        /// </summary>
+        private bool translateModification;
+
+        /// <summary>
+        /// The game object with mesh that the user has chosen that will be placed.
+        /// </summary>
+        private GameObject gameObject;
+
+        /// <summary>
+        /// The original position of a building that will be moved.
+        /// </summary>
+        private Vector3 previousPosition;
+
+        /// <summary>
         /// Initializes a new instance of the <see cref="ObjectPlacementState"/> class.
         /// </summary>
         /// <param name="stateContext">State context</param>
         /// <param name="location">First indicated position of the placement</param>
-        public ObjectPlacementState(StateContext stateContext, Vector3 location) : base(stateContext)
+        /// <param name="gameObject">The game object that has to be placed or moved</param>
+        public ObjectPlacementState(StateContext stateContext, Vector3 location, GameObject gameObject) : base(stateContext)
+        {
+            gameObject.SetActive(true);
+            this.translateModification = gameObject.GetComponent<MetaBody>() != null;
+            if (this.translateModification)
+            {
+                this.previousPosition = gameObject.transform.position;
+                UnityEngine.Object.Destroy(gameObject.GetComponent<MetaBody>());
+                UnityEngine.Object.Destroy(gameObject.GetComponent<GroundPlane>());
+            }
+
+            this.hoverTime = StopwatchSingleton.Time;
+            this.gameObject = gameObject;
+            this.StateContext.Buttons.BackButton.SetActive(true);
+            this.gameObject.GetComponent<Renderer>().material.SetColor("_Color", Color.green);
+            this.gameObject.transform.position = location;
+        }
+
+        /// <summary>
+        /// Shows the position of the to be placed object and places it after having hovered for 3 seconds.
+        /// </summary>
+        /// <param name="position">position of the to be placed building</param>
+        public override void OnPoint(Vector3 position)
+        {
+            long time = StopwatchSingleton.Time;
+            if ((position - this.gameObject.transform.position).magnitude > (position.magnitude / 10f))
+            {
+                this.gameObject.GetComponent<Renderer>().material.SetColor("_Color", Color.yellow);
+                this.hoverTime = time;
+            }
+            else
+            {
+                this.gameObject.GetComponent<Renderer>().material.SetColor("_Color", Color.green);
+            }
+
+            this.gameObject.transform.position = position;
+            if (time - this.hoverTime > 2500)
```

Comment:
There is unnecessary in this if statement. Might also be a good idea to create some private methods for parts, since the method is quite long, hard to understand and seems to be doing multiple things.